Straight_skeleton_extrusion_2/include/CGAL/extrude_skeleton.h

```diff
@@ -0,0 +1,1137 @@
+// Copyright (c) 2023 GeometryFactory
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Mael Rouxel-Labbé
+//
+
+#ifndef CGAL_SLS_EXTRUSION_EXTRUDE_SKELETON_H
+#define CGAL_SLS_EXTRUSION_EXTRUDE_SKELETON_H
+
+#include <CGAL/license/Straight_skeleton_extrusion_2.h>
+
+#ifdef CGAL_SLS_DEBUG_DRAW
+  #include <CGAL/draw_straight_skeleton_2.h>
+  #include <CGAL/draw_polygon_2.h>
+  #include <CGAL/draw_polygon_with_holes_2.h>
+  #include <CGAL/draw_triangulation_2.h>
+#endif
+
+#include <CGAL/create_weighted_straight_skeleton_2.h>
+#include <CGAL/create_weighted_straight_skeleton_from_polygon_with_holes_2.h>
+#include <CGAL/create_weighted_offset_polygons_2.h>
+#include <CGAL/create_weighted_offset_polygons_from_polygon_with_holes_2.h>
+#include <CGAL/Straight_skeleton_2/Straight_skeleton_aux.h>
+
+#include <CGAL/General_polygon_with_holes_2.h>
+
+#include <CGAL/Polygon_mesh_processing/border.h>
+#include <CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h>
+#include <CGAL/Polygon_mesh_processing/repair_polygon_soup.h>
+#include <CGAL/Polygon_mesh_processing/stitch_borders.h>
+
+#ifdef CGAL_SLS_OUTPUT_FILES
+#include <CGAL/IO/polygon_soup_io.h>
+#endif
+
+#include <CGAL/Constrained_Delaunay_triangulation_2.h>
+#include <CGAL/Projection_traits_3.h>
+#include <CGAL/Triangulation_vertex_base_with_info_2.h>
+#include <CGAL/mark_domain_in_triangulation.h>
+
+#include <CGAL/Cartesian/function_objects.h>
+#include <CGAL/enum.h>
+#include <CGAL/IO/helpers.h>
+#include <CGAL/Kernel_traits.h>
+#include <CGAL/Polygon_2.h>
+
+#include <boost/algorithm/clamp.hpp>
+#include <boost/optional/optional.hpp>
+#include <boost/range/value_type.hpp>
+#include <boost/shared_ptr.hpp>
+
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <type_traits>
+#include <unordered_map>
+#include <vector>
+
+namespace CGAL {
+namespace Straight_skeleton_extrusion {
+namespace internal {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// Below is to handle vertical slabs.
+
+enum class Slope
+{
+  UNKNOWN = 0,
+  INWARD,
+  OUTWARD,
+  VERTICAL
+};
+
+template <typename FT>
+const FT default_extrusion_height = { (std::numeric_limits<double>::max)() };
```

Comment:
In fact this seems to be the culprit for the non-empty memory pool.

Reply:
I tried this 
```
template <typename FT>
struct Default_extrusion_height {
    CGAL_STATIC_THREAD_LOCAL_VARIABLE(FT, value, { (std::numeric_limits<double>::max)() });
};
```
but this does not compile.  I guess it does not like the initializer list in the macro.

Reply:
If I put this line in the free functions that need the variable it works. 
```
CGAL_STATIC_THREAD_LOCAL_VARIABLE(FT, default_extrusion_height, DBL_MAX);
``` 
@MaelRL @sloriot let me know if I shall commit such a change.